src/reference/constellation/index.pug

```diff
@@ -0,0 +1,160 @@
+
+    +highlightStr('json','{"type": "event", "event": "math_result", "data": 4}')
+    ul
+        li #[strong type] is always set to "event"
+        li #[strong event] is the string name of the event
+        li #[strong data] is unstructured information associated with the event. Usually this is the same as found within the liveloading/sails event of the same type.
+    h2#errors Error Codes
+    p.
+        Constellation uses the 4xxx error code range reserved for use by applications in addition to the standard 1xxx codes. The following codes are in use:
+    ul
+        li #[strong 1011] - Sent in a close or method reply if an unknown internal error occurs.
+        li #[strong 1012] - Sent in a close frame when we deploy or restart Constellation; clients should attempt to reconnect.
+        li #[strong 4000] - Error parsing payload as JSON
+        li #[strong 4001] - Error decompressing a supposedly-gzipped payload
+        li #[strong 4002] - Unknown packet type
+        li #[strong 4003] - Unknown method name call
+        li #[strong 4004] - Error parsing method arguments (not the right type or structure)
+        li #[strong 4005] - The user session has expired; if using a cookie, they should log in again, or get a bearer auth token if using an authorization header.
+        li #[strong 4100] - Unknown event in a livesubscribe call
+        li #[strong 4101] - You do not have access to subscribe to the livesubscribe event
+        li #[strong 4102] - You are already subscribed to the livesubscribe event (during livesubscribe)
+        li #[strong 4103] - You are not subscribe to the livesubscribe event (during liveunsubscribe)
+    h2#messages Messages
+    h3#messages_hello hello
+    p.
+        A hello event is sent down to the client when they first connect.
+    +highlightStr('json','{"type": "event", "event": "hello", "data": {"authenticated": true}}')
+    ul
+        li authenticated is true if the socket consumer authenticated as a user rather than a guest
+    h3#messages_livesubscribe livesubscribe
+    p.
+        A livesubscribe method is made available for users to listen to liveloading events.
+
+    +highlightStr('json','{"type": "method", "method": "livesubscribe", "params": {"events": ["user:1:update", "channel:1:followed"]}, "id": 42}')
+    ul
+        li  events is an array of events to subscribe to. Note that either all events are successfully subscribe to, or a failure occurs and no events are subscribed to. Either do or do not, there is no try.
+    p.
+        A livesubscribe reply looks like one of the following:
+    ul
+        li.
+            A successful reply:
+        +highlightStr('json','{"type": "reply", "result": null, "error": null, "id": 42}')
+        li.
+            A reply with an invalid event:
+        +highlightStr('json','{"type": "reply", "result": null, "error": {"code": 4100, "message": "Unknown event \'my silly event\' "}, "id": 42}')
```

Comment:
yep, that's right 😛 

![](https://peet.io/i/16-08-r9nww5leatw38d6fqbteze8arc37b5.png)